opgee/processes/acid_gas_removal.py

```diff
@@ -61,6 +62,21 @@ class AcidGasRemoval(Process):
     def __init__(self, name, **kwargs):
         super().__init__(name, **kwargs)
 
+        self._required_inputs = [
+            "gas for AGR",          # TODO: avoid process names in contents. Should be "acidic gas"?
+        ]
+
+        self._required_outputs = [
+            # TODO: If the process name were avoided, we could have just one output stream
+            #  with, say, "deacidified gas". Should describe the contents, not the destination.
+            # One of these must exist.
+            ("gas for demethanizer",  # TODO: avoid process names in contents
+             "gas for gas partition") # TODO: avoid process names in contents
+        ]
+
+        # Optional streams include:
+        # "gas for CO2 compressor" (output)
+
```

Comment:
Regarding your comments, I agree that uniquely identifying the specific names seems brittle. My initial thought is that the declarations would lay out the number of streams and their contents, maybe with an optional identifier. What if we had something like:
```python
Phase = Literal['gas', 'solid', 'liquid']
class ProcessIOConstraint(NamedTuple):
    gas: int|List[Phase]
    oil: int
...
self._required_inputs: ProcessIOConstraints = ProcessIOConstraints(gas: 1)
```
There are likely ways of indicating "1 or more", "1 and only 1", or ">1,<=3", but it seems like the count and some kind of contents info would be necessary without using dynamic input values.

Reply:
I'll have to think about this further, but my first impression is that this proposal is more complicated than necessary, and fails to address the original intent of the content strings.

The point of the stream "content" string is to allow processes to identify expected streams so they know which ones to read from and write to as required. (Before this, Processes would traverse the Stream to identify the Process on the other end.)

Anyway, I think a simple string accomplishes the original goal with minimal fuss. Having the validation code verify that the expected contents are present strikes me as the right level of validation.

Although the goal of avoiding dependencies on specific network structure in the Process subclass code was subverted by the inclusion of Process names in content strings, we can fix that pretty easily by renaming these.

Reply:
It's already merged, but the main concern I have is that those names are set in the XML. Choosing a different name for a stream would then fail validation while potentially being a valid field. l think we'd want to either decouple the streams/processes from the input or remove the stream naming from the input.